### PR TITLE
Add dialog accept and cancel events

### PR DESCRIPTION
**Pull request by plusCubed**
https://github.com/material-components/material-components-web/tree/master/packages/mdc-dialog#dialog-events

### CODE REVIEW

File: Dialog/Dialog.jsx [hunk 1]
```diff
@@ -11,10 +11,24 @@ class Dialog extends MaterialComponent {
   }
   componentDidMount() {
     this.MDComponent = new MDCDialog(this.control);
+    this.MDComponent.listen("MDCDialog:accept", this._onAccept);
```
Comment (prateekbh) [inline Dialog/Dialog.jsx]: you need to to bind it with `this` in the constructor

File: Dialog/Dialog.jsx [hunk 2]
```diff
@@ -8,13 +8,29 @@ class Dialog extends MaterialComponent {
...
+    this._onAccept = this._onAccept.bind(this);
+    this._onCancel = this._onCancel.bind(this);
   }
   componentDidMount() {
     this.MDComponent = new MDCDialog(this.control);
+    this.MDComponent.listen("MDCDialog:accept", this._onAccept);
+    this.MDComponent.listen("MDCDialog:cancel", this._onCancel);
   }
   componentWillUnmount() {
+    this.MDComponent.unlisten("MDCDialog:accept", this._onAccept);
```
Comment (prateekbh) [inline Dialog/Dialog.jsx]: `this.MDComponent.unlisten && this.MDComponent.unlisten(...`

Comment (plusCubed) [inline Dialog/Dialog.jsx]: The method will always exist, no? I mean we could check `this.MDComponent` instead, if that's necessary

Comment (prateekbh) [inline Dialog/Dialog.jsx]: no so its not a public method so far, hence the check

Comment (plusCubed) [inline Dialog/Dialog.jsx]: https://material.io/components/web/catalog/base/#methods?

Comment (prateekbh) [inline Dialog/Dialog.jsx]: oh sry my bad, got confused

**Pull request merged**